Parallel GH actions workflow for Nixpkgs eval (#356023)
NixOS/nixpkgs: ci/eval/README.md

# Nixpkgs CI evaluation

> The code in this directory is used by the [eval.yml](../../.github/workflows/eval.yml) GitHub Actions workflow to evaluate the majority of Nixpkgs for all PRs, effectively making sure that when the development branches are processed by Hydra, it doesn't encounter any evaluation failures.

Reviewer: ```suggestion
The code in this directory is used by the [eval.yml](../../.github/workflows/eval.yml) GitHub Actions workflow to evaluate the majority of Nixpkgs for all PRs, effectively making sure that when the development branches are processed by Hydra, they don't encounter any evaluation failures.
```
Is "branches" or "Hydra" the subject?

Author: Meant for it to be Hydra, how about avoiding the ambiguity by getting rid of the thing that could be ambiguous haha:
```suggestion
The code in this directory is used by the [eval.yml](../../.github/workflows/eval.yml) GitHub Actions workflow to evaluate the majority of Nixpkgs for all PRs, effectively making sure that when the development branches are processed by Hydra, no evaluation failures are encountered.
```

Reviewer: Yep, your suggestion looks good to me.

Furthermore it also allows local evaluation using
```
nix-build ci -A eval.full \
  --max-jobs 4
  --cores 2
  --arg chunkSize 10000
```

- `--max-jobs`: The maximum number of derivations to run at the same time. Only each [supported system](../supportedSystems.nix) gets a separate derivation, so it doesn't make sense to set this higher than that number.
- `--cores`: The number of cores to use for each job. Recommended to set this to the amount of cores on your system divided by `--max-jobs`.

> - `chunkSize`: The number of attributes that are evaluated simultaneously on a single core. Lowering this decreases memory usage at the cost of increased evaluation time. If this it too high, there won't be enough chunks to process them in parallel, also increasing evaluation time.

Reviewer: ```suggestion
- `chunkSize`: The number of attributes that are evaluated simultaneously on a single core. Lowering this decreases memory usage at the cost of increased evaluation time. If this is too high, there won't be enough chunks to process them in parallel, and will also increase evaluation time.
```
